internal/notifier/factory.go

```diff
@@ -85,6 +85,8 @@ func (f Factory) Notifier(provider string) (Interface, error) {
 		n, err = NewGitLab(f.ProviderUID, f.URL, f.Token, f.CertPool)
 	case apiv1.GiteaProvider:
 		n, err = NewGitea(f.ProviderUID, f.URL, f.Token, f.CertPool)
+	case apiv1.BitbucketServerProvider:
+		n, err = NewBitbucketServer(f.ProviderUID, f.URL, f.Token, f.CertPool, f.Username, f.Password)
```

Comment:
The primary reason I dismissed the idea was that I was following pattern for existing Git Commit Status providers as much as possible. Since none of the existing ones reference it, I thought it was not necessary for this one as well. 

If that's not the case, I don't mind implementing it and testing it. Please let me know!

Reply:
I have not implemented proxy support yet. Bitbucket Server/Data Center is a self hosted offering of Atlassian Bitbucket and is typically used by big enterprises . Proxy is usually required by enterprises to provide access to external services. Since the target that this provider caters to will be internally hosted, having proxy support for this provider seems unnecessary. Nevertheless, if you guys believe its required, I can submit a follow-up change to add support for it. Thx.